Shared/sdk/SharedUtil.Hash.hpp

```diff
@@ -818,5 +818,22 @@ namespace SharedUtil
 
         out->assign((char*)buffer, numPasses * 4);
         delete[] buffer;
+
+        if (trimPaddingAtEnd)
+        {
+            // Check last block for 0 padding, and delete
+
+            auto iter = out->end() - 1;
```

Comment:
What if `out.begin() == out.end()` (aka. the input is empty)?

Reply:
The block size is 4. If its smaller than that the function returns before ever reaching this point.